### PR TITLE
[rt-thread][doc] update rt-thread supported BSPs (up to 21 BSPs)

### PR DESCRIPTION
**Pull request by mysterywolf**
### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Checkpoints
- [ ] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [ ] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [ ] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [ ] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [ ] Use typed pointers instead of `void *` pointers
- [ ] Do not `malloc` into a static or global variables. Instead declare the variable in `LV_ITERATE_ROOTS` list in [`lv_gc.h`](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_gc.h) and mark the variable with `GC_ROOT(variable)` when it's used. See a detaild description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [ ] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [ ] Widget members function must start with `lv_<modul_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [ ] `struct`s should be used via an API and not modified directly via their elements.
- [ ] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [ ] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [ ] Arguments must be named in H files too.
- [ ] To register and use callbacks one of the followings needs to be followed (see a detaild description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)): 
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

File: docs/get-started/os/rt-thread.md [hunk 1]
```diff
@@ -17,29 +17,25 @@
...
-| [Nuvoton nk-n9h30](https://github.com/RT-Thread/rt-thread/tree/master/bsp/nuvoton/nk-n9h30/applications/lvgl) |      |
-| [Nuvoton numaker-m032ki](https://github.com/RT-Thread/rt-thread/tree/master/bsp/nuvoton/numaker-m032ki/applications/lvgl) |      |
-| [NXP imxrt1060-evk](https://github.com/RT-Thread/rt-thread/tree/master/bsp/imxrt/imxrt1060-nxp-evk/applications/lvgl) |      |
-| [STM32L475 pandora](https://github.com/RT-Thread/rt-thread/tree/master/bsp/stm32/stm32l475-atk-pandora/applications/lvgl) |      |
-| [STM32F407 explorer](https://github.com/RT-Thread/rt-thread/tree/master/bsp/stm32/stm32f407-atk-explorer/applications/lvgl) |      |
-| [STM32F469 Discovery](https://github.com/RT-Thread/rt-thread/tree/master/bsp/stm32/stm32f469-st-disco/applications/lvgl) |      |
-| [Raspberry PICO](https://github.com/RT-Thread/rt-thread/tree/master/bsp/raspberry-pico/applications/lvgl) |      |
-
+| BSP                                                                                                                               | Note                                                                                                                                              |
```
Comment (kisvegabor) [inline docs/get-started/os/rt-thread.md:26]: It seems there are BSPs in both columns.

```suggestion
| BSP                                                                                                                               | BSP                                                                                                                                              |
```

**Comment by kisvegabor** [comment]
Wow, great news!

File: docs/get-started/os/rt-thread.md (hunk 1 above)
Comment (mysterywolf) [reply docs/get-started/os/rt-thread.md:26]: Yes, I intentionally did that. Because there is no note for BSPs, and there are more BSPs supported.

Comment (kisvegabor) [reply docs/get-started/os/rt-thread.md:26]: I don't get it. It seems there are the same kind of BSPs in both columns: links to 
`https://github.com/RT-Thread/rt-thread/tree/master/bsp/<TARGET>/applications/lvgl`, but the first column is called BSP, and the second is `Note`

Comment (mysterywolf) [reply docs/get-started/os/rt-thread.md:26]: I leave a blank for Note, but there is note for each bsp in fact. So the note can be removed. Two column are all BSPs to reduce the rows of the document. Just like this: https://github.com/RTduino/RTduino#11-rt-thread-bsps-which-have-supported-rtduino

Comment (kisvegabor) [reply docs/get-started/os/rt-thread.md:26]: > I leave a blank for Note

Sounds good!

Comment (mysterywolf) [reply docs/get-started/os/rt-thread.md:26]: emm no, I mean, I leaved a blank for note for every BSP, but I found that those blank notes are meaningless, so I remove it now.

Comment (mysterywolf) [reply docs/get-started/os/rt-thread.md:26]: Do you want to leave that notes? or we can remove those

Comment (kisvegabor) [reply docs/get-started/os/rt-thread.md:26]: I still don't get it, but I'm sure you know now better why it should be this way.  :slightly_smiling_face: 

So I merge it.

**Comment by mysterywolf** [comment]
Thanks~

**Pull request merged**